src/test/java/how/hollow/ExampleTest.java

```diff
@@ -0,0 +1,148 @@
+
+    announceNewVersion(latestVersion, new HollowWriteStateEngineBuilder()
+        .add("some irrelevant data")
+        .add(movie)
+        .build());
+
+    MoviePrimaryKeyIndex index = new MoviePrimaryKeyIndex(consumer);
+    assertEquals(movie.title, index.findMatch(movie.id).getTitle().getValue());
+  }
+
+  @Test
+  public void actorIdLookup() throws IOException {
+    Actor actor = new Actor(2, "actor name");
+    Movie movie = new Movie(1, "title", Set.of(actor));
+
+    announceNewVersion(latestVersion, new HollowWriteStateEngineBuilder()
+        .add(movie)
+        .add(actor)
+        .build());
+
+    ActorPrimaryKeyIndex index = new ActorPrimaryKeyIndex(consumer);
+    assertEquals(actor.actorName, index.findMatch(actor.actorId).getActorName().getValue());
+  }
+
+  @Test
+  public void actorIdLookupTransitively() throws IOException {
+    Actor actor = new Actor(2, "actor name");
+    Movie movie = new Movie(1, "title", Set.of(actor));
+
+    announceNewVersion(latestVersion, new HollowWriteStateEngineBuilder()
+        .add(movie)
+        .build());
+
+    ActorPrimaryKeyIndex index = new ActorPrimaryKeyIndex(consumer);
+    assertEquals(actor.actorName, index.findMatch(actor.actorId).getActorName().getValue());
+  }
+
+  @Test
+  public void actorHashIndexLookup() throws IOException {
+    Actor actor = new Actor(2, "actor name");
+    Movie movie = new Movie(1, "title", Set.of(actor));
+
+    announceNewVersion(latestVersion, new HollowWriteStateEngineBuilder()
+        .add(movie)
+        .build());
+
+    MovieAPIHashIndex index = new MovieAPIHashIndex(consumer, "Movie", "",
+        "actors.element.actorName.value");
+    Iterator<how.hollow.consumer.api.generated.Movie> result = index.findMovieMatches("actor name")
+        .iterator();
+    assertTrue(result.hasNext());
+    assertEquals(movie.title, result.next().getTitle().getValue());
+    assertFalse(result.hasNext());
+    assertFalse(index.findMovieMatches("other name").iterator().hasNext());
+  }
+
+  @Test
+
+  public void actorHashIndexLookupReturnsEmptyIterator() throws IOException {
+    Movie movie = new Movie(1, "title", Set.of());
+
+    announceNewVersion(latestVersion, new HollowWriteStateEngineBuilder()
+        .add(movie)
+        .build());
+
+    MovieAPIHashIndex index = new MovieAPIHashIndex(consumer, "Movie", "",
+        "actors.element.actorName.value");
+
+    Iterator<how.hollow.consumer.api.generated.Movie> result = index.findMovieMatches("other name")
+        .iterator();
+    assertFalse(result.hasNext());
+  }
+
+
+  @Test
+  public void movieIndexUpdate() throws Exception {
+
+    Movie movie = new Movie(1, "title", Set.of());
+    Movie newMovie = new Movie(2, "coming soon", Set.of());
+
+    announceNewVersion(latestVersion, new HollowWriteStateEngineBuilder()
+        .add(movie)
+        .build());
+
+    MoviePrimaryKeyIndex index = new MoviePrimaryKeyIndex(consumer);
```

Comment:
Isn't this deprecated now?

In my example I can see that I could basically use something like :
```
// Setting up an id index
val uniqueIndex = Movie.uniqueIndex(consumer)
consumer.addRefreshListener(uniqueIndex)
```
The `Movie` type is the generated class.

Reply:
Yes, I'll fix that as soon as I have your PR merged and then I can use the new API.